e2e/tests/negative/test_backup_listing.robot

```diff
@@ -0,0 +1,180 @@
+    Write 210 GB large data to file 0 in pod 2
+
+Write 150 GB data in pod 4
+    Write 150 GB large data to file 0 in pod 4
+
+Write 40 GB data in pod 4
+    Write 40 GB large data to file 1 in pod 4
+
+Create pod ${pod_id} from volume ${source_volume_id} ${source_volume_size} GB volume backup ${backup_id}
+    ${source_volume_name}=    generate_name_with_suffix    volume    ${source_volume_id}
+    ${backup_url}=    get_backup_url    ${backup_id}    ${source_volume_name}
+    ${volume_name}=    generate_name_with_suffix    volume    ${pod_id}
+    create_volume   ${volume_name}    size=${source_volume_size}Gi    numberOfReplicas=3    fromBackup=${backup_url}    dataEngine=${DATA_ENGINE}
+    Create persistentvolume for volume ${pod_id}
+    Create persistentvolumeclaim for volume ${pod_id}
+    Create pod ${pod_id} using volume ${pod_id}
+    Wait for pod ${pod_id} running
+
+Delete pod ${pod_id} and volume ${volume_id}
+    Delete pod ${pod_id}
+    Wait for volume ${volume_id} detached
+    Delete volume ${volume_id}
+
+Pod ${pod_id} data should same as volume ${source_volume_id} backup ${backup_id}
+    ${source_volume_name}=    generate_name_with_suffix    volume    ${source_volume_id}
+    ${backup_name}=    get_backup_name    ${backup_id}    ${source_volume_name}
+    ${expected_checksum}=    get_restored_checksum    ${backup_name}
+    ${pod_name} =    generate_name_with_suffix    pod    ${pod_id}
+    ${current_checksum}=    get_workload_pod_data_checksum    ${pod_name}    0
+    Should Be Equal  ${expected_checksum}  ${current_checksum}
+    ...  msg="expected ${expected_checksum}, got ${current_checksum}!"
+
+*** Test Cases ***
+Backup listing with more than 1000 backups
+    [Tags]  manual  longhorn-8355
+    [Documentation]    Test backup listing
+    Given Create persistentvolumeclaim 0 using RWO volume
+    And Create deployment 0 with persistentvolumeclaim 0
+    And Write data to file in deployment 0
+    Then Perform backup 1001 times for deployment 0 volume
+    Then Create volume 1 from deployment 0 volume random backup
+    And Create deployment 1 with volume 1
+    Then Get deployment 1 volume data in file data
+    And Volume 1 data should same as deployment 0 volume
+
+Backup listing of volume bigger than 200 Gi
+    [Tags]  manual  longhorn-8355
+    [Documentation]    Test backup bigger than 200 Gi
+    Given Create persistentvolumeclaim 0 using RWO volume
+    And Create deployment 0 with persistentvolumeclaim 0
+    And Write data to file in deployment 0
+    Then Perform backup 1001 times for deployment 0 volume
+    Then Create volume 1 from deployment 0 volume random backup
+    And Create deployment 1 with volume 1
+    Then Get deployment 1 volume data in file data
+    And Volume 1 data should same as deployment 0 volume
+    Then Create pod 2 mount 250 GB volume 2
+    And Write 210 GB data in pod 2
+    Then Volume 2 backup 0 should be able to create
+    Then Delete pod 2 and volume 2
```

Comment:
Is the deletion necessary here? Can we handle pod and volume deletion within the teardown cleanup function instead?

Reply:
In this case, we need to create a 250GB `Volume 2` and then create `Volume 3` by `restoring from the backup of Volume 2`. In the test environment where I develop test cases, each worker node has a 1TB hard drive. However, I haven't confirmed the minimum resource requirements for this case. If we want to use a shared cluster for each case, I believe interspersing deletions in the case store can help us avoid using excessively large disk resources.